azurerm/resource_arm_front_door.go

```diff
@@ -0,0 +1,1459 @@
+				Backends:              expandArmFrontDoorBackend(backends),
+				LoadBalancingSettings: expandArmFrontDoorSubResource(subscriptionId, resourceGroup, frontDoorName, "LoadBalancingSettings", backendPoolLoadBalancingName),
+				HealthProbeSettings:   expandArmFrontDoorSubResource(subscriptionId, resourceGroup, frontDoorName, "HealthProbeSettings", backendPoolHealthProbeName),
+			},
+		}
+
+		output = append(output, result)
+	}
+
+	return &output
+}
+
+func expandArmFrontDoorBackend(input []interface{}) *[]frontdoor.Backend {
+	if len(input) == 0 {
+		return &[]frontdoor.Backend{}
+	}
+
+	output := make([]frontdoor.Backend, 0)
+
+	for _, be := range input {
+		backend := be.(map[string]interface{})
+
+		address := backend["address"].(string)
+		hostHeader := backend["host_header"].(string)
+		enabled := backend["enabled"].(bool)
+		httpPort := int32(backend["http_port"].(int))
+		httpsPort := int32(backend["https_port"].(int))
+		priority := int32(backend["priority"].(int))
+		weight := int32(backend["weight"].(int))
+
+		result := frontdoor.Backend{
+			Address:           utils.String(address),
+			BackendHostHeader: utils.String(hostHeader),
+			EnabledState:      expandArmFrontDoorBackendEnabledState(enabled),
+			HTTPPort:          utils.Int32(httpPort),
+			HTTPSPort:         utils.Int32(httpsPort),
+			Priority:          utils.Int32(priority),
+			Weight:            utils.Int32(weight),
+		}
+
+		output = append(output, result)
+	}
+
+	return &output
+}
+
+func expandArmFrontDoorBackendEnabledState(isEnabled bool) frontdoor.BackendEnabledState {
+	if isEnabled {
+		return frontdoor.Enabled
+	}
+
+	return frontdoor.Disabled
+}
+
+func expandArmFrontDoorBackendPoolsSettings(enforceCertificateNameCheck bool) *frontdoor.BackendPoolsSettings {
+	enforceCheck := frontdoor.EnforceCertificateNameCheckEnabledStateDisabled
+
+	if enforceCertificateNameCheck {
+		enforceCheck = frontdoor.EnforceCertificateNameCheckEnabledStateEnabled
+	}
+
+	result := frontdoor.BackendPoolsSettings{
+		EnforceCertificateNameCheck: enforceCheck,
+	}
+
+	return &result
+}
+
+func expandArmFrontDoorFrontendEndpoint(input []interface{}, subscriptionId string, resourceGroup string, frontDoorName string) *[]frontdoor.FrontendEndpoint {
+	if len(input) == 0 {
+		return &[]frontdoor.FrontendEndpoint{}
+	}
+
+	output := make([]frontdoor.FrontendEndpoint, 0)
+
+	for _, frontendEndpoints := range input {
+		frontendEndpoint := frontendEndpoints.(map[string]interface{})
+
+		hostName := frontendEndpoint["host_name"].(string)
+		isSessionAffinityEnabled := frontendEndpoint["session_affinity_enabled"].(bool)
+		sessionAffinityTtlSeconds := int32(frontendEndpoint["session_affinity_ttl_seconds"].(int))
+		customHttpsConfiguration := frontendEndpoint["custom_https_configuration"].([]interface{})
+		name := frontendEndpoint["name"].(string)
+		id := utils.String(helper.GetFrontDoorSubResourceId(subscriptionId, resourceGroup, frontDoorName, "FrontendEndpoints", name))
```

Comment:
And then this could become:
```suggestion
		id := utils.String(frontDoorID + "/FrontendEndpoints/" + name))
```

Reply:
Fixed